### PR TITLE
must have full path to bash, otherwise fails (on Arch)

### PR DESCRIPTION
**Pull request by gdamjan**

### CODE REVIEW

**Comment by jfrazelle** [comment]
Can you please sign your commits following these rules:

https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work

The easiest way to do this is to amend the last commit:

`git commit --amend -s --no-edit && git push -f`

**Comment by gdamjan** [comment]
is it good now?

**Comment by jfrazelle** [comment]
yes all good :D

**Comment by jfrazelle** [comment]
@tianon is there anything that could go wrong with this?

**Comment by tianon** [comment]
I use this script all the time and have never run into an issue with it.  Can you elaborate on your environment?

**Comment by gdamjan** [comment]
I've digged a bit more into it.

Arch doesn't really have /bin or /sbin (they are symlinks now, per the /usr unification thing [1]) so it doesn't have them in the PATH either. the PATH variable is inherited from the host shell to the commands running in the chroot container. So it is `/usr/local/sbin:/usr/local/bin:/usr/bin` when run from default Arch system.

So one approach would be to specify full path to executables, and in the case of `bash` I would say it's better to use `/bin/bash`, but also that script should maybe normalize PATH to what Debian expects for normal operation (i.e. `export PATH=/usr/local/sbin:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin`).

[1]
http://lwn.net/Articles/483921/

**Comment by tianon** [comment]
Ah, thanks for digging; that makes more sense!  In that case, I think I'd rather see us actually `export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin` somewhere in this script, especially to make sure that even though we've specified the full path to bash, we can't control things that `apt-get` might run, so I think we should be more defensive and provide the environment they're expecting.

**Comment by crosbymichael** [comment]
LGTM

ping @tianon

File: contrib/mkimage/debootstrap [hunk 1]
```diff
@@ -189,9 +189,10 @@ fi
 
 (
 	set -x
+	export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
 	
 	# make sure we're fully up-to-date
-	chroot "$rootfsDir" bash -c 'apt-get update && apt-get dist-upgrade -y'
+	chroot "$rootfsDir" /bin/bash -c 'apt-get update && apt-get dist-upgrade -y'
```
Comment (tianon) [inline contrib/mkimage/debootstrap]: Doesn't the full `PATH` mean we don't need this explicit `/bin/` anymore?

**Comment by tianon** [comment]
FYI @gdamjan it doesn't notify us when you've updated the PR, so feel free to poke :+1:

File: contrib/mkimage/debootstrap (hunk 1 above)
Comment (gdamjan) [inline contrib/mkimage/debootstrap]: true. 
I usually do prefer explicit paths for various reasons, but in this case it's the least of issues.

**Comment by jfrazelle** [comment]
@gdamjan are you able to make these changes otherwise one of us can carry the PR to make sure they get in :)

**Comment by jfrazelle** [comment]
I believe this also fixes https://github.com/docker/docker/issues/8258 right?

**Comment by tianon** [comment]
@jfrazelle indeed it does!

**Comment by tianon** [comment]
I'm going to carry this one from here.

**Pull request closed without merging**